modules/RTC/RTC.js

```diff
@@ -143,6 +220,150 @@ export default class RTC extends Listenable {
     }
 
     /**
+     * Gets streams from specified device types and creates {@code
+     * JitsiLocalTrack} instances for each. This call intentionally ignores
+     * errors for upstream to catch and handle instead.
+     *
+     * @param {Object} options - A hash describing what devices to get and
+     * relevant constraints.
+     * @param {string[]} options.devices - The types of media to capture. Valid
+     * values are "desktop", "audio", and "video".
+     * @returns {Promise} The promise, when successful, will return an array of
+     * JitsiLocalTrack instances. If an error occurs, it will be deferred to the
+     * caller for handling.
+     */
+    static _newObtainAudioAndVideoPermissions(options = {}) {
+        logger.info('Using the new gUM flow');
+
+        const localJitsiTracks = [];
+
+        // Declare private functions to be used in the promise chain below.
+        // These functions are declared in the scope of this function because
+        // they are not being used anywhere else, so only this function needs to
+        // know about them.
+
+        /**
+         * Executes a request for desktop media if specified in options.
+         *
+         * @returns {Promise}
+         */
+        function maybeRequestDesktopDevice() {
+            const devices = options.devices || [ 'audio', 'video' ];
+            const requestedDesktopDevices = devices.filter(
+                device => device === 'desktop');
+
+            return requestedDesktopDevices.length
+                ? RTCUtils._newGetDesktopMedia(
+                    options.desktopSharingExtensionExternalInstallation,
+                    options.desktopSharingSources)
+                : Promise.resolve();
+        }
+
+        /**
+         * Instantiates a {@code JitsiLocalTrack} with the passed in desktop
+         * stream and pushes the model instance to the internal array
+         * localJitsiTracks to be returned later.
+         *
+         * @param {MediaStreamTrack} desktopStream - A track for a desktop
+         * capture.
+         * @returns {void}
+         */
+        function maybeCreateDesktopModel(desktopStream) {
```

Comment:
i find 'model' here to be a bit confusing.  what about `maybeCreateAndAddDesktopTrack`?